File: Auto/Java01/src/com/company/Auto.java
package com.company;

public class Auto {
    //Atributos
    private String Patente;
    private int cantRuedas;
    private String color;
    private String marca;
    private String modelo;
    private boolean tieneAireAcondicionado;
    private float cilindrada;
    private double precio;
    private char categoria;


    //Constructores
    //Constructor por defecto
    public Auto() {
    }

    //Constructor por parametros n1
    public Auto(String Patente, int cantRuedas, String color, String marca, boolean tieneAireAcondicionado, float cilindrada, double precio, char categoria) {
        this.Patente = Patente;
        this.cantRuedas = cantRuedas;
        this.color = color;
        this.marca = marca;
        this.tieneAireAcondicionado = tieneAireAcondicionado;
        this.cilindrada = cilindrada;
        this.precio = precio;
        this.categoria = categoria;
    }
    //Constructor por parametros n2
    public Auto(int cantRuedas, String color, String marca, boolean tieneAireAcondicionado, float cilindrada, double precio, char categoria) {
    }

    //Getters
    public String getPatente() {
        return Patente;
    }
    public int getCantRuedas() {
        return cantRuedas;
    }
    public String getColor() {
        return color;
    }
    public String getMarca() {
        return marca;
    }
    public String getModelo() {
        return modelo;
    }
    public boolean getTieneAireAcondicionado() {
        return tieneAireAcondicionado;
    }
    public float getCilindrada() {
        return cilindrada;
    }
    public double getPrecio() {
        return precio;
    }
    public char getCategoria(){ return categoria;}

    //Setters
    public void setPatente(String Patente) {
        this.Patente = Patente;
    }
    public void setCantRuedas(int cantRuedas) {
        this.cantRuedas = cantRuedas;
    }
    public void setColor(String color) {
        this.color = color;
    }
    public void setMarca(String marca) {
        this.marca = marca;
    }
    public void setModelo(String modelo) {
        this.modelo = modelo;
    }
    public void setTieneAireAcondicionado(boolean tieneAireAcondicionado) {
        this.tieneAireAcondicionado = tieneAireAcondicionado;
    }
    public void setCilindrada(float cilindrada) {
        this.cilindrada = cilindrada;
    }
    public void setPrecio(double precio) {
        this.precio = precio;
    }

    //Metodos
    private void encender() {
        System.out.println("El auto se encendió");
    }
    public void frenar() {
        System.out.println("El auto freno");
    }
    public void acelerar() {
        System.out.println("El auto acelero");
    }
    public void doblarDerecha() {
        System.out.println("El auto giro a la derecha");
    }
    public void doblarIzquierda() {
        System.out.println("El auto giro a la izquierda");
    }

    @Override
    public String toString() {
        return "Auto{" + "Patente=" + Patente + ", cantRuedas=" + cantRuedas + ", color=" + color + ", marca=" + marca +
                ", modelo=" + modelo + ", tieneAireAcondicionado=" + tieneAireAcondicionado + ", cilindrada=" + cilindrada + ", precio=" + precio + '}';
    }
}

File: Auto/Java01/src/com/Auto/JS01/clases/Auto.js
class Auto {
    //Métodos
    //Constructores
    //Constructor por defecto
    Auto() {
    }
    //Constructor por parametros n1
    Auto(patente, cantRuedas, color, marca, modelo, tieneAireAcondicionado, cilindrada, precio, categoria) {
        this.patente = patente;
        this.cantRuedas = cantRuedas;
        this.color = color;
        this.marca = marca;
        this.modelo = modelo;
        this.tieneAireAcondicionado = tieneAireAcondicionado;
        this.cilindrada = cilindrada;
        this.precio = precio;
        this.categoria = categoria;
    }
    //Constructor por parametros n2
    Auto(patente, cantRuedas, color, precio, categoria) {
        this.patente = patente;
        this.cantRuedas = cantRuedas;
        this.color = color;
        this.precio = precio;
        this.categoria = categoria;
    }

    //Getters
    getPatente() {
        return this.patente;
    }
    getCantRuedas() {
        return this.cantRuedas;
    }
    getColor() {
        return this.color;
    }
    getMarca() {
        return this.marca;
    }
    getModelo() {
        return this.modelo;
    }
    getTieneAireAcondicionado() {
        return this.tieneAireAcondicionado;
    }
    getCilindrada() {
        return this.cilindrada;
    }
    getPrecio() {
        return this.precio;
    }
    getCategoria() {
        return this.categoria;
    }

    //Setters
    setPatente(patente) {
        this.patente = patente;
    }
    setCantRuedas(cantRuedas) {
        this.cantRuedas = cantRuedas;
    }
    setColor(color) {
        this.color = color;
    }
    setMarca(marca) {
        this.marca = marca;
    }
    setModelo(modelo) {
        this.modelo = modelo;
    }
    setTieneAireAcondicionado(tieneAireAcondicionado) {
        this.tieneAireAcondicionado = tieneAireAcondicionado;
    }
    setCilindrada(cilindrada) {
        this.cilindrada = cilindrada;
    }
    setPrecio(precio) {
        this.precio = precio;
    }
    setCategoria(categoria) {
        this.categoria = categoria;
    }


    //toString
    toString() {
        return "Patente: " + this.patente + " Cantidad de Ruedas: " + this.cantRuedas + " Color: " + this.color + " Marca: " + this.marca + " Modelo: " + this.modelo + " Tiene Aire Acondicionado: " + this.tieneAireAcondicionado + " Cilindrada: " + this.cilindrada + " Precio: " + this.precio + " Categoria: " + this.categoria;
    }

}

//Crear auto
let ferrari01 = new Auto("ABC123", 4, "Rojo", "Ferrari", "Testa Rossa", true, 1.6, 100000, "A");
let datosDeLaferrari = ferrari01.toString();

console.log(datosDeLaferrari);
